Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -26,35 +26,77 @@
 #include "config.h"
 #include "CookieStore.h"
 
+#include "Cookie.h"
 #include "CookieInit.h"
+#include "CookieJar.h"
+#include "CookieListItem.h"
 #include "CookieStoreDeleteOptions.h"
 #include "CookieStoreGetOptions.h"
+#include "Document.h"
+#include "JSCookieListItem.h"
 #include "JSDOMPromiseDeferred.h"
+#include "ScriptExecutionContext.h"
 #include <wtf/IsoMallocInlines.h>
 #include <wtf/Ref.h>
+#include <wtf/Vector.h>
 #include <wtf/text/WTFString.h>
 
 namespace WebCore {
 
 WTF_MAKE_ISO_ALLOCATED_IMPL(CookieStore);
 
-Ref<CookieStore> CookieStore::create()
+Ref<CookieStore> CookieStore::create(Document* document)
 {
-    return adoptRef(*new CookieStore());
+    auto cookieStore = adoptRef(*new CookieStore(document));
+    cookieStore->suspendIfNeeded();
+    return cookieStore;
 }
 
-CookieStore::CookieStore() = default;
+CookieStore::CookieStore(Document* document)
+    : ActiveDOMObject(document)
+{
+}
 
 CookieStore::~CookieStore() = default;
 
-void CookieStore::get(const String&, Ref<DeferredPromise>&& promise)
+void CookieStore::get(String&& name, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    get(CookieStoreGetOptions { WTFMove(name), { } }, WTFMove(promise));
 }
 
-void CookieStore::get(CookieStoreGetOptions&&, Ref<DeferredPromise>&& promise)
+void CookieStore::get(CookieStoreGetOptions&& options, Ref<DeferredPromise>&& promise)
 {
-    promise->reject(NotSupportedError);
+    auto* context = scriptExecutionContext();
+    if (!context) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    auto* origin = context->securityOrigin();
+    if (!origin) {
+        promise->reject(SecurityError);
+        return;
+    }
+
+    if (origin->isOpaque()) {
+        promise->reject(Exception { SecurityError, "The origin is opaque"_s });
+        return;
+    }
+
+    auto& url = context->url();
```

Comment:
Could you please move this to the next line and call document.url() instead? It would avoid a virtual function call